csaf-validation/src/main/kotlin/io/github/csaf/sbom/validation/roles/Roles.kt

```diff
@@ -0,0 +1,113 @@
+import io.github.csaf.sbom.validation.requirements.Requirement15
+import io.github.csaf.sbom.validation.requirements.Requirement16
+import io.github.csaf.sbom.validation.requirements.Requirement17
+import io.github.csaf.sbom.validation.requirements.Requirement18
+import io.github.csaf.sbom.validation.requirements.Requirement19
+import io.github.csaf.sbom.validation.requirements.Requirement20
+import io.github.csaf.sbom.validation.requirements.Requirement21
+import io.github.csaf.sbom.validation.requirements.Requirement22
+import io.github.csaf.sbom.validation.requirements.Requirement23
+import io.github.csaf.sbom.validation.requirements.Requirement5
+import io.github.csaf.sbom.validation.requirements.Requirement6
+import io.github.csaf.sbom.validation.requirements.Requirement7
+import io.github.csaf.sbom.validation.requirements.Requirement8
+import io.github.csaf.sbom.validation.requirements.Requirement9
+import io.github.csaf.sbom.validation.requirements.TlpWhiteAccessible
+import io.github.csaf.sbom.validation.requirements.UsageOfTls
+import io.github.csaf.sbom.validation.requirements.ValidCSAFDocument
+import io.github.csaf.sbom.validation.requirements.ValidFilename
+import io.github.csaf.sbom.validation.requirements.YearInFolder
+
+/**
+ * The "CSAF publisher" role. See
+ * https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#721-role-csaf-publisher.
+ */
+object CSAFPublisherRole : Role {
+
+    override val roleRequirements = none()
+
+    override val documentRequirements =
+        allOf(ValidCSAFDocument, ValidFilename, UsageOfTls, TlpWhiteAccessible)
+}
+
+/**
+ * The "CSAF provider" role. See
+ * https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#722-role-csaf-provider.
+ */
+object CSAFProviderRole : Role {
+    override val roleRequirements =
+        CSAFPublisherRole.roleRequirements +
+            allOf(Requirement6, Requirement7) +
+            oneOf(Requirement8, Requirement9, Requirement10) +
+            (allOf(YearInFolder, Requirement12, Requirement13, Requirement14) or
+                allOf(Requirement15, Requirement16, Requirement17))
+
+    override val documentRequirements = CSAFPublisherRole.documentRequirements + Requirement5
+}
+
+/**
+ * The "CSAF trusted provider role".
+ * https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#723-role-csaf-trusted-provider.
+ */
+object CSAFTrustedProviderRole : Role {
+    override val roleRequirements = CSAFProviderRole.roleRequirements + Requirement20
+
+    override val documentRequirements =
+        CSAFProviderRole.documentRequirements + allOf(Requirement18, Requirement19)
+}
+
+/**
+ * The "CSAF lister role". See
+ * https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#724-role-csaf-lister.
+ */
+object CSAFListerRole : Role {
+    override val roleRequirements: Requirement = allOf(Requirement6, Requirement21, Requirement22)
+
+    override val documentRequirements = none()
```

Comment:
These two requirements seem to be missing:

> * uses the value lister for /aggregator/category.
> * does not list any mirror pointing to a domain under its own control.

Reply:
This specification really is "meh" at some point.. Why is there no dedicated number for these requirements.. Let's deal with them later.